packages/esm-patient-medications-app/src/add-drug-order/drug-order-form.component.tsx

```diff
@@ -165,6 +178,22 @@ export function DrugOrderForm({ initialOrderBasketItem, onSave, onCancel }: Drug
       startDate: defaultStartDate,
     },
   });
+  const {
+    handleSubmit,
+    control,
+    watch,
+    setValue,
+    formState: { errors },
+  } = methods;
+
+  const handleUnitAfterChange = (
+    newValue: MedicationOrderFormData['unit'],
+    prevValue: MedicationOrderFormData['unit'],
+  ) => {
+    if (prevValue?.valueCoded === watch('quantityUnits')?.valueCoded) {
```

Comment:
This can probably be substituted (more correctly) with `getValues()` (also returned from `useForm()`. `watch()` is intended to subscribe to input changes, but here it's used to get a point-in-time value.

Reply:
Since this is passed as a prop to a component, it would be best to wrap this in `useCallback()`. Otherwise, each render will re-render the `ControlledFieldInput`.

Reply:
So, I don't think the logic here avoids the edge-case where the user has selected a `quantityUnits` value that happens to be the same as the `dosingUnits`, but then changes the `dosingUnits`. I'm ok with automatically changing the value as long as its been automatically determined, but user input (especially for medication instructions) needs to always override automatic determinations.